### ISSUE TITLE
Bad performance when pasting long strings

### ISSUE THREAD

**Issue by lukaskollmer** [omz/Pythonista-Issues#146]
When I paste a long string in Pythonista, the app freezes for a couple of seconds (~5-10). This happens both for pasting in the editor and the console. 

However, when I paste the same string in another app, like Drafts or Safari, it is pasted instantly and there is no lag at all. 

Example string to reproduce: https://gist.github.com/lukaskollmer/ec703b8d6daacda5dfd16c6faa8a8af4

**Comment by cclauss** [comment]
You will see the same effect if you paste that text into a Python file in Atom.  The reason is because these code editors (unlike Drafts or Safari) are trying to do code highlighting and your string is causing them to churn thru a lot of characters without finding a space or recognizable Python syntax.

**Comment by Phuket2** [comment]
@cclauss, I agree. @lukaskollmer , I used your long string. When I tried to paste it into a .py file, after a min or more I gave up.  Surprised you could get 10 secs. But if you paste it into a text file in Pythonista it's instant. 
I copied your text to the clipboard and tried -
editor.make_new_file('clip.py', clipboard.get())
No difference, still a long time. Was just a thought.

**Comment by zrzka** [comment]
Not sure if it's worth to invest time into this one.

* When pasted into the txt file, it's instant (just tested in 311014)
* When pasted into the py file, it takes up to 10s

Why this happens already described by @cclauss. Something like this doesn't belong to py files. Should be saved as another file and loaded in the Python script.

I would close this one. @omz any plans to work on syntax highlighting performance?
